scylla/src/transport/connection.rs

```diff
@@ -273,14 +282,29 @@ impl NonErrorQueryResponse {
             }
         };
 
-        Ok(QueryResult {
-            rows,
-            warnings: self.warnings,
-            tracing_id: self.tracing_id,
-            paging_state_response: paging_state,
-            col_specs,
-            serialized_size,
-        })
+        Ok((
+            QueryResult {
+                rows,
+                warnings: self.warnings,
+                tracing_id: self.tracing_id,
+                col_specs,
+                serialized_size,
+            },
+            paging_state,
+        ))
+    }
+
+    pub(crate) fn into_query_result(self) -> Result<QueryResult, QueryError> {
+        let (result, paging_state) = self.into_query_result_and_paging_state()?;
+
+        if !paging_state.finished() {
+            let error_msg = "Internal driver API misuse or a server bug: nonfinished paging state\
+                             discarded by `NonErrorQueryResponse::into_query_result`";
+            error!(error_msg);
+            return Err(QueryError::ProtocolError(error_msg));
+        }
```

Comment:
`discarded by` doesn't sound right now that the whole query is errored out.
`passed to`?

Reply:
`would be discarded` -> this is the idea. Does it sound good enough?

Reply:
ok